### PR TITLE
MMT-976 add select2 field for instrument short name and populate long…

### PR DESCRIPTION
**Pull request by macrouch**
… name

### CODE REVIEW

File: app/assets/javascripts/select2.coffee [hunk 1]
```diff
@@ -148,3 +148,31 @@ $(document).ready ->
...
+    longName = $element.find(':selected').data('longName')
+
+    $longNameElement = $element.parent().siblings().find('.instrument-long-name')
+
+    if longName?
+      $longNameElement.val(longName)
+      $longNameElement.attr('readonly', true)
+    else
+      $longNameElement.val('')
+      $longNameElement.attr('readonly', false)
```
Comment (charleshuang80) [inline app/assets/javascripts/select2.coffee]: So we have MMT-1116, that a user shouldn't be able to key in a longname for KMS values. Should that apply to instruments as well?

Comment (macrouch) [reply app/assets/javascripts/select2.coffee]: Yes it should, but this ticket it written this way. I like the idea of taking care of all of the long name fields at the same time anyway

Comment (charleshuang80) [reply app/assets/javascripts/select2.coffee]: Taking care of it all in 1116 seems fine to me, but it looks like there are comments in the ticket that seem contrary to how long name behavior is written in the acceptance criteria, just fyi.

**Comment by charleshuang80** [comment]
Should the Instrument Child fields be covered by this?

**Pull request merged**